.github/workflows/release-build-publish.yml

```diff
@@ -0,0 +1,256 @@
+        run: |
+          # Remove the tests directory from the package
+          TEST_DIR=$(find . -type d -name "tests")
+          if [ -n "$TEST_DIR" ]; then
+            rm -rf $TEST_DIR
+          else
+            echo "No tests directory found! This is unexpected."
+            exit 1
+          fi
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.11
+
+      - name: Install dependencies
+        run: pip install --upgrade setuptools wheel twine
+
+      - name: Perform some tests on the metadata of the package
+        run: |
+          python setup.py check --metadata --strict
+
+      - name: Build package
+        run: |
+          python setup.py sdist bdist_wheel
+
+      - name: Test package
+        run: |
+          twine check dist/*
+
+      - name: Verify contents of the tar.gz
+        env:
+          CLEANLAB_PACKAGE_NAME: ${{ needs.extract_project_name.outputs.cleanlab_package_name }}
+        run: |
+          # Extract the tar.gz to a temporary directory
+          TMPDIR=$(mktemp -d)
+          tar -xzf dist/*.tar.gz -C $TMPDIR
+          PACKAGE_DIR=$TMPDIR/$(ls $TMPDIR) # Assuming there's only one directory extracted
+
+          # Define expected files and directories
+          EXPECTED_FILES="LICENSE PKG-INFO MANIFEST.in DEVELOPMENT.md CONTRIBUTING.md CODE_OF_CONDUCT.md README.md pyproject.toml setup.cfg setup.py"
+          EXPECTED_DIRS="cleanlab $CLEANLAB_PACKAGE_NAME.egg-info"
+
+          # Collect actual top-level files and directories
+          ACTUAL_CONTENTS=$(ls $PACKAGE_DIR)
+          ACTUAL_FILES=$(find $PACKAGE_DIR -maxdepth 1 -type f -exec basename {} \;)
+          ACTUAL_DIRS=$(find $PACKAGE_DIR -maxdepth 1 -type d -exec basename {} \; | sed "1d") # Remove the package directory itself from the list
+
+          # Function to check for unexpected or missing files and directories
+          check_contents() {
+            local missing=0
+            # Check for expected files
+            for expected_file in $EXPECTED_FILES; do
+              echo $ACTUAL_FILES | grep -qw $expected_file || { echo "Missing expected file: $expected_file"; missing=1; }
+            done
+            
+            # Check for expected directories
+            for expected_dir in $EXPECTED_DIRS; do
+              echo $ACTUAL_DIRS | grep -qw $expected_dir || { echo "Missing expected directory: $expected_dir"; missing=1; }
+            done
+
+            # Check for unexpected files and directories
+            for actual in $ACTUAL_CONTENTS; do
+              echo $EXPECTED_FILES $EXPECTED_DIRS | grep -qw $actual || { echo "Unexpected item in package: $actual"; missing=1; }
+            done
+
+            # Exit if anything unexpected or missing
+            if [ $missing -ne 0 ]; then
+              echo "Package content verification failed."
+              exit 1
+            fi
+          }
+
+          # Execute checks
+          check_contents
```

Comment:
This is the best way I found to check that there are no unanticipated files added in the distribution, nor are there any missing files or top-level directories.

Something like this is good:

```bash
❯ tree -L 1 cleanlab-2.5.0
cleanlab-2.5.0
├── CODE_OF_CONDUCT.md
├── CONTRIBUTING.md
├── DEVELOPMENT.md
├── LICENSE
├── MANIFEST.in
├── PKG-INFO
├── README.md
├── cleanlab
│   └── ...
├── cleanlab.egg-info
│   └── ...
├── pyproject.toml
├── setup.cfg
└── setup.py
```